lib/formtastic/inputs/check_boxes_input.rb

```diff
@@ -180,11 +180,18 @@ def checkbox_input(choice)
       def make_selected_values
         if object.respond_to?(method)
           selected_items = object.send(method)
-
           # Construct an array from the return value, regardless of the return type
           selected_items = [*selected_items].compact.flatten
 
-          [*selected_items.map { |o| send_or_call_or_object(value_method, o) }].compact
+          selected = []
+          selected_items.each do |o|
+            item = send_or_call_or_object(value_method, o)
+            if item == nil
+              item = o.id if o.respond_to? :id
+            end
+            selected << item
+          end
+          selected
```

Comment:
Yes, I will test these things out. I also need to add some test coverage for HABTM. I got the tests running on my machine by running bundle exec rspec but bundle exec rake still fails.

@justinfrench you seem to be heavily mocking and stubbing in your tests. Do you have an example of how you would want a HABTM test setup? I can make something from scratch but if you have a helper method or pattern you want followed it would make life a little easier.